### DISCUSSION TITLE
Distributing Cython code

### DISCUSSION THREAD

**Discussion by erwanp** [radis/radis#238] [Q&A]
_(trying this GitHub Discussion feature to see if it's a relevant channel, half-way between Issues and [our Slack](https://radis.github.io/slack-invite/) )_ 

With #234  @dcmvdbekerom introduced new Cython code. 

Questions are : 

1. How do we make sure it's compiled properly on the test suite ([Travis-CI](https://travis-ci.com/github/radis/radis) ) ? Some example projects I've found simply add the  `python setup.py build_ext` line in their [.travis.yml](https://github.com/theochem/python-cython-ci-example/blob/master/.travis.yml#L139) 

---

2. How do we make sure that radis users can use the Cython version without having to type another install ? (radis is currently installed with `pip install radis`, this shouldn't change ! [unless we move to Conda?] ) . Some ideas : 
- currently, @dcmvdbekerom introduced a command line call in a [try/except](https://github.com/radis/radis/pull/234/files#diff-85a035414f3d48ffc0db8079be5cac5553de6dd376e4844da9917ffa289dfb64R105) block directly at runtime. A bit hacky but works !
- [Cython documentation](https://cython.readthedocs.io/en/latest/src/quickstart/build.html#building-a-cython-module-using-setuptools) seems to suggest to add the Cythonize steps directly in `setup.py`. Is this equivalent to the former build command ?
- other ideas or experiences ?

---

3. [Cython documentation](https://cython.readthedocs.io/en/latest/src/quickstart/build.html#building-a-cython-module-using-setuptools) suggests to also deploy the compiled C code, so people do not have to install Cython and can use their default C. I can find recent StackOverflow discussions where this is debatted.   Should we include these files ? 
- To me it seems that Radis should anyway be installed on top of Anaconda, where Cython is installed by default. This should simplify everything !
- Some people, however, still install RADIS with `virtuaenv` ([radis-app](https://github.com/suzil/radis-ui) runs this way). We may make sure there is still a pure-python default option (slower), but without C-code anyway. If people want C-speed everywhere they get Anaconda & Cython.

**Comment by gully** [comment]
I saw this talk [Cython for All with GitHub Actions](https://www.youtube.com/watch?v=-7_07O5ENhU) at the SF Python Meetup in 2019.  I think it is describing the solution you seek--- having GitHub Actions (or equivalent) automatically build the various platform-specific wheels and send them to pypi.
